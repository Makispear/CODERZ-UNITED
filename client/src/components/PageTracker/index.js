export default function PageTracker({ props }) {

  return (
    <span data-testid="pageTracker" className="text-tertiary p-style">{props.page}</span>
  )
}
