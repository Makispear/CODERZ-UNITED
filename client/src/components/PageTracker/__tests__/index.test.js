import PageTracker from "..";
import { render, cleanup } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

afterEach(cleanup)

const pageNumber = "1.1.1"

describe("'PageTracker' Component", () => {
    it('renders', () => {
        render(<PageTracker props={{ page: pageNumber }} />);
    });

    it("has classNames 'text-tertiary p-style'", () => {
        const { getByTestId } = render(<PageTracker props={{ page: pageNumber }} />)
        const testId = getByTestId('pageTracker')

        expect(testId).toHaveClass('text-tertiary p-style')
    })

    it('Matches snapshot DOM node structure', () => {
        const { asFragment } = render(<PageTracker props={{ page: pageNumber }} />);
        expect(asFragment()).toMatchSnapshot();
    });

    // it('has 3 options "Design, Front-End & Back-End"', () => {
    //     const { getByTestId } = render(<Options />)

    //     expect(getByTestId('design')).toHaveTextContent(('design').toUpperCase())
    //     expect(getByTestId('frontEnd')).toHaveTextContent(('< frontEnd >'))
    //     expect(getByTestId('backEnd')).toHaveTextContent(('back_end'))
    // })
})