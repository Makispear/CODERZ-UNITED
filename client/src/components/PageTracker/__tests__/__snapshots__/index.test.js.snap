// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'PageTracker' Component Matches snapshot DOM node structure 1`] = `
<DocumentFragment>
  <span
    class="text-tertiary p-style"
    data-testid="pageTracker"
  >
    1.1.1
  </span>
</DocumentFragment>
`;
